app/controllers/help_controller.rb

```diff
@@ -89,4 +89,8 @@ def catch_spam
     end
   end
 
+  def recaptcha_required?
```

Comment:
I'm not sure about this implementation. [The work required on the theme end](https://github.com/mysociety/whatdotheyknow-theme/pull/521/files#diff-c38892457e3c9bd7f3c03c9d1a066150R20) is quite complex.

The [filter](https://github.com/mysociety/whatdotheyknow-theme/pull/521/files#diff-c38892457e3c9bd7f3c03c9d1a066150R20) should exist in core, and we should use `@recaptcha_required` to decide whether to show and validate the recaptcha. At the moment, the theme has to care about setting [`@recaptcha_required`](https://github.com/mysociety/whatdotheyknow-theme/pull/521/files#diff-c38892457e3c9bd7f3c03c9d1a066150R34) _and_ overriding [`recaptcha_required?`](https://github.com/mysociety/whatdotheyknow-theme/pull/521/files#diff-c38892457e3c9bd7f3c03c9d1a066150R37).

```ruby
# Core
class HelpController < ApplicationController
  before_filter :set_recaptcha_required, only: [:contact]

  def contact
    # …
    if (@recaptcha_required && !params[:remove] && !verify_recaptcha)
      flash.now[:error] = _('There was an error with the reCAPTCHA. ' \
                            'Please try again.')
    elsif @contact.valid? && !params[:remove]
    # …
  end

  private

  def set_recaptcha_required
    @recaptcha_required = AlaveteliConfiguration.contact_form_recaptcha
  end
end
```

That way, the only thing a theme owner needs to do is:

* Set `CONTACT_FORM_RECAPTCHA` to `true`
* Add recaptcha to the form:

```erb
<% if @recaptcha_required %>
  <%= recaptcha_tags %>
<% end %>
```

If a theme owner wanted to only show for requests from foreign IPs, the only extra thing they need to do is set `@recaptcha_required` appropriately, without having to worry about adding filters etc.

```ruby
# Additional checks in theme
HelpController.class_eval do
  def set_recaptcha_required
    @recaptcha_required =
      AlaveteliConfiguration.contact_form_recaptcha &&
      request_from_foreign_country?
  end

  def request_from_foreign_country?
    country_from_ip != AlaveteliConfiguration.iso_country_code
  end
end
```

Reply:
Good point - I was too focused on this exact case rather than the simplest possible override scenario. Thanks!